test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,860 @@
+                expect(consoleSpy.getCall(0).firstArg).to.equal(
+                    Messages.InformationMessages.StartingEndSession
+                );
+
+                // Should only log a single verbose log
+                expect(consoleSpy.getCalls().length).to.equal(1);
+            });
+
+            it('should log a NoSessionToEnd Message if Persistence Exists but does not return an sid', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and NoSessionToEnd messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.NoSessionToEnd
+                );
+            });
+
+            it('should log an AbandonedEndSession message if unable to log', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                sinon.stub(mpInstance._Helpers, 'canLog').returns(false);
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and AbandonEndSession messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.AbandonEndSession
+                );
+            });
+
+            it('should log an AbandonedEndSession message if Store is not enabled', () => {
+                mParticle.init(apiKey, window.mParticle.config);
+
+                const mpInstance = mParticle.getInstance();
+                sinon
+                    .stub(mpInstance._Persistence, 'getPersistence')
+                    .returns({ gs: {} });
+
+                const consoleSpy = sinon.spy(mpInstance.Logger, 'verbose');
+
+                mpInstance._Store.isEnabled = false;
+
+                mpInstance._SessionManager.endSession();
+
+                // Should log initial StartingEndSession and AbandonEndSession messagesk
+                expect(consoleSpy.getCalls().length).to.equal(2);
+                expect(consoleSpy.lastCall.firstArg).to.equal(
+                    Messages.InformationMessages.AbandonEndSession
+                );
+            });
+
+            // QUESTION: What exactly defines an "AbandonedEndSession"?
```

Comment:
Searching through out codebase, we have things similar messages like `AbandonLogEvent`, `AbandonAliasUsers`, `AbandonStartSession`, and `AbandonEndSession`.  They all happen if logging is disabled or developer token is not sent.  

Although there is also `LoggingDisabled`, which we should audit and determine what the difference is.